lsp/angularls.lua

```diff
@@ -0,0 +1,87 @@
+-- Angular requires a node_modules directory to probe for @angular/language-service and typescript
+-- in order to use your projects configured versions.
+local root_dir = vim.fn.getcwd()
+local project_root = vim.fs.dirname(vim.fs.find('node_modules', { path = root_dir, upward = true })[1])
+
+local function get_probe_dir()
+  return project_root and (project_root .. '/node_modules') or ''
+end
+
+local function get_angular_core_version()
+  if not project_root then
+    return ''
+  end
+
+  local package_json = project_root .. '/package.json'
+  if not vim.uv.fs_stat(package_json) then
+    return ''
+  end
+
+  local contents = io.open(package_json):read '*a'
+  local json = vim.json.decode(contents)
+  if not json.dependencies then
+    return ''
+  end
+
+  local angular_core_version = json.dependencies['@angular/core']
+
+  angular_core_version = angular_core_version and angular_core_version:match('%d+%.%d+%.%d+')
+
+  return angular_core_version
+end
+
+local default_probe_dir = get_probe_dir()
+local default_angular_core_version = get_angular_core_version()
```

Comment:
evaluating this here (and getcwd() above) means that if the user changes the CWD later, that won't be used by the config.

It seems like `cwd` should be defined as a function for this config, instead of a list. That makes it more "dynamic".

The old config https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/configs/angularls.lua had this issue too, though, so I guess this can be revisited later...